distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso

```diff
@@ -388,3 +344,78 @@ is_valid v = case v of
     Nothing -> False
     _ : Number -> v.is_nan.not
     _ -> True
+
+## PRIVATE
+type Accumulator
+    new : Integer -> Accumulator
+    new moments=0 = Accumulator.Value 0 Nothing Nothing (Vector.fill moments 0)
+
+    ## PRIVATE
+    Value count:Integer minimum:Any maximum:Any (moments:Vector|Nothing)
+
+    ## PRIVATE
+    comparator_error : Boolean
+    comparator_error self = self.minimum.is_nothing
+
+    ## PRIVATE
+    increment self value = if self.count == 0 then (Accumulator.Value 1 value value (self.update_moments value)) else
+        minimum = Accumulator.comparator self.minimum value Ordering.Less
+        maximum = Accumulator.comparator self.maximum value Ordering.Greater
+        Accumulator.Value self.count+1 minimum maximum (self.update_moments value)
+
+    ## PRIVATE
+    comparator current value bound = if current == Nothing then Nothing else
```

Comment:
Again a bit of a nitpick, but the code would be a bit easier to understand with clearer names or some comments what the function is supposed to do.

Why not call this method `compare` or `perform_comparison`? Something with an action-verb. `comparator` sounds like it is generating a comparator instance, not performing the actual comparison.

Reply:
The name of the argument `bound` is also very unclear - of course reading the code deeply it can be inferred, but at a glance it's not clear what that is.